### PR TITLE
feat(argo-workflows): Add global tag

### PR DESCRIPTION
**Pull request by yu-croco**
I added `.Values.images.tag` 🙋 

---

Signed-off-by: yu-croco <yuki.kita22@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-workflows/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: argo-workflows
 description: A Helm chart for Argo Workflows
 type: application
-version: 0.16.7
+version: 0.17.0
```
Comment (yu-croco) [inline charts/argo-workflows/Chart.yaml:5]: I removed `.Values. images` , so I upgraded minor version. 
I am wondering if I should keep `.Values.images` as deprecated... 🤔

File: charts/argo-workflows/values.yaml [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
-  pullPolicy: Always
-  # -- Secrets with credentials to pull images from a private registry
-  pullSecrets: []
-  # - name: argo-pull-secret
+global:
+  image:
+    # -- Overrides the global Argo Workflows image tag whose default is the chart appVersion
+    tag: ""
+    # -- If defined, a imagePullPolicy applied to all Argo Workflows deployments
+    pullPolicy: IfNotPresent
```
Comment (yu-croco) [inline charts/argo-workflows/values.yaml:6]: I think `IfNotPresent` is better, compared to `Always`...? 🤔

**Comment by vladlosev** [comment]
This change will break setups for many users who rely on the values in question. They will suddenly find that they are running a different (possibly, very different) version of Argo Workflows. This is a big price to pay, but the value of the improvement it is buying is questionable.

**Comment by yu-croco** [comment]
@vladlosev 
Thank you for your opinion! I agree with it.

Keeping `.Values.images` as deprecated would be tough (at least some breaking change would happen), so I stop moving them to `global.image` .
Instead, may I just add `tag` on `.Values.images.tag` ? 👀 
At a minimum, what I would like to improve is to add `tag` attribute as global.

**Comment by vladlosev** [comment]
`.Values.images.tag` as another default will not be a breaking change and work fine.

**Comment by yu-croco** [comment]
@vladlosev 
> .Values.images.tag as another default will not be a breaking change and work fine.

Thank you! I added `.images.tag`. 🙋

File: charts/argo-workflows/README.md [hunk 3]
```diff
@@ -45,6 +45,7 @@ Fields to note:
 | fullnameOverride | string | `nil` | String to fully override "argo-workflows.fullname" template |
 | images.pullPolicy | string | `"Always"` | imagePullPolicy to apply to all containers |
 | images.pullSecrets | list | `[]` | Secrets with credentials to pull images from a private registry |
+| images.tag | string | `""` | Overrides the global Argo Workflows image tag whose default is the chart appVersion |
```
Comment (vladlosev) [inline charts/argo-workflows/README.md:48]: Common tag for Argo Workflows images. Defaults to `.Chart.AppVersion.`.

File: charts/argo-workflows/README.md [hunk 4]
```diff
@@ -73,7 +74,7 @@ Fields to note:
 | controller.extraEnv | list | `[]` | Extra environment variables to provide to the controller container |
 | controller.image.registry | string | `"quay.io"` | Registry to use for the controller |
 | controller.image.repository | string | `"argoproj/workflow-controller"` | Registry to use for the controller |
-| controller.image.tag | string | `""` | Overrides the image tag whose default is the chart appVersion. |
+| controller.image.tag | string | `""` | Overrides the image tag whose default is `images.tag`. |
```
Comment (vladlosev) [inline charts/argo-workflows/README.md:77]: Image tag for the workflow controller. Defaults to `.Values.images.tag`.

File: charts/argo-workflows/README.md [hunk 5]
```diff
@@ -143,7 +144,7 @@ Fields to note:
 | executor.env | object | `{}` | Adds environment variables for the executor. |
 | executor.image.registry | string | `"quay.io"` | Registry to use for the Workflow Executors |
 | executor.image.repository | string | `"argoproj/argoexec"` | Repository to use for the Workflow Executors |
-| executor.image.tag | string | `""` | Overrides the image tag whose default is the chart appVersion. |
+| executor.image.tag | string | `""` | Overrides the image tag whose default is `images.tag`. |
```
Comment (vladlosev) [inline charts/argo-workflows/README.md:147]: Image tag for the workflow executor. Defaults to `.Values.images.tag`.

File: charts/argo-workflows/README.md [hunk 6]
```diff
@@ -162,7 +163,7 @@ Fields to note:
 | server.extraEnv | list | `[]` | Extra environment variables to provide to the argo-server container |
 | server.image.registry | string | `"quay.io"` | Registry to use for the server |
 | server.image.repository | string | `"argoproj/argocli"` | Repository to use for the server |
-| server.image.tag | string | `""` | Overrides the image tag whose default is the chart appVersion. |
+| server.image.tag | string | `""` | Overrides the image tag whose default is `images.tag`. |
```
Comment (vladlosev) [inline charts/argo-workflows/README.md:166]: Image tag for the Argo Workflows server. Defaults to `.Values.images.tag`.

**Review by vladlosev — COMMENTED** [review]
The changes look fine, but I suggest slightly different wordings in the documentation, for clarity.

**Comment by yu-croco** [comment]
@vladlosev 
Thank you for reviewing! I fixed them. 🙋

**Review by vladlosev — APPROVED** [review]
LGTM

**Comment by yu-croco** [comment]
Thank you for reviewing!

**Pull request merged**